applications/GeoMechanicsApplication/custom_workflows/dgeosettlement.cpp

```diff
@@ -59,6 +106,15 @@ int KratosGeoSettlement::RunStage(const std::filesystem::path&            rWorki
         KRATOS_INFO("KratosGeoSettlement") << "Read the materials from " << material_file_path << std::endl;
     }
 
+    std::vector<std::shared_ptr<Process>> processes = GetProcesses(project_parameters);
+    std::vector<std::weak_ptr<Process>> process_observables;
+    process_observables.insert(process_observables.end(), processes.begin(), processes.end());
+
+    if (mpTimeLoopExecutor != nullptr)
```

Comment:
@avdg81 I added this nullptr check, because when calling the API from python now, there is no implementation yet for the time loop executor, so it will be a nullptr. If we want to test this a bit through python, we don't want it to segfault here, since it's only temporary anyway. Later on, we should move this check to the constructor, same as with the InputUtility

Reply:
The check is fine with me. I would only express it slightly differently:
```suggestion
    if (mpTimeLoopExecutor)
```

Reply:
Changed